tests/Tests/ORM/Hydration/ComplexeObjectHydratorTest.php

```diff
@@ -0,0 +1,111 @@
+use Doctrine\Tests\Models\ComplexeHydration\Serie;
+use Doctrine\Tests\Models\ComplexeHydration\SerieImportator;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function count;
+use function dirname;
+use function substr;
+
+/**
+ * @requires PHP 8.1
+ */
+class ComplexeObjectHydratorTest extends OrmFunctionalTestCase
+{
+    /** @var bool */
+    private $entitiesCreated = false;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->_em         = $this->getEntityManager(null, new AttributeDriver([dirname(__DIR__, 2) . '/Models/ComplexeHydration'], true));
+        $this->_schemaTool = new SchemaTool($this->_em);
+
+        if (! $this->entitiesCreated) {
+            $this->entitiesCreated = true;
+
+            $this->setUpEntitySchema([Serie::class, Item::class, SerieImportator::class, ItemSerie::class]);
+
+            for ($iSerie = 1; $iSerie <= 2; $iSerie++) {
+                $serie = new Serie();
+                $serie->setLibelle('Serie ' . $iSerie);
+                $this->_em->persist($serie);
+
+                $serieImportator = new SerieImportator();
+                $serieImportator->setLibelle('imp ' . $iSerie);
+                $serieImportator->setSerie($serie);
+                $this->_em->persist($serieImportator);
+
+                for ($iItem = 1; $iItem <= 12; $iItem++) {
+                    $item = new Item();
+                    $item->setLibelle('Item ' . $iSerie . '.' . $iItem);
+
+                    $itemSerie = new ItemSerie();
+                    $itemSerie->setSerie($serie);
+                    $itemSerie->setItem($item);
+                    $itemSerie->setNumber('Num ' . $iSerie . '.' . $iItem);
+
+                    $this->_em->persist($item);
+                    $this->_em->persist($itemSerie);
+                }
+            }
+
+            $this->_em->flush();
+            $this->_em->clear();
+        }
+    }
+
+    protected function tearDown(): void
+    {
+        $this->_em->createQuery('DELETE FROM ' . ItemSerie::class)->execute();
+        $this->_em->createQuery('DELETE FROM ' . SerieImportator::class)->execute();
+        $this->_em->createQuery('DELETE FROM ' . Serie::class)->execute();
+        $this->_em->createQuery('DELETE FROM ' . Item::class)->execute();
+    }
+
+    protected function getDQL(): string
+    {
+        return 'SELECT s0_, i1_, i2_, s4_, s5_, i6_ FROM ' . Serie::class . ' s0_ LEFT JOIN s0_.itemSeries i1_ LEFT JOIN i1_.item i2_ LEFT JOIN s0_.serieImportators s4_ LEFT JOIN s4_.serie s5_ LEFT JOIN s5_.itemSeries i6_ ORDER BY i1_.serie ASC, i1_.item ASC';
+    }
+
+    public function testComplexEntityQueryInObject(): void
+    {
+        $dql  = $this->getDQL();
+        $data = $this->_em->createQuery($dql)->getResult();
+
+        self::assertEquals(2, count($data));
+
+        for ($i = 0; $i < 2; $i++) {
+            foreach ($data[$i]->getItemSeries()->toArray() as $element) {
```

Comment:
Maybe you could use something more concrete than `$element` here? `$itemSerie` maybe?